### PR TITLE
Update pytorch code to LazyAPI.

### PR DESCRIPTION
**Pull request by cheungdaven**
*Description of changes:*
This commit updated the pytorch api to lazyapi, up to the attention chapter.

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Comment by cheungdaven** [comment]
@d2l-bot please rebuild.

File: chapter_convolutional-modern/alexnet.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-```{.python .input}
+```{.python .input  n=1}
```
Comment (astonzhang) [inline chapter_convolutional-modern/alexnet.md:1]: plz remove n=XX by running "d2lbook activate all"

File: chapter_convolutional-modern/batch-norm.md [hunk 2]
```diff
@@ -622,7 +624,9 @@ while our custom implementation must be interpreted by Python.
 %%tab mxnet, pytorch
 trainer = d2l.Trainer(max_epochs=10, num_gpus=1)
 data = d2l.FashionMNIST(batch_size=256)
-model = BNLeNet(lr=0.9)
+model = BNLeNet(lr=0.1)
```
Comment (astonzhang) [inline chapter_convolutional-modern/batch-norm.md:627]: Why do we change the lr?

**Comment by astonzhang** [comment]
To exclude earlier changes (init fix) in this PR, plz rebase this PR onto the master branch.

**Comment by cheungdaven** [comment]
@astonzhang The commits passed the CI (from scratch) and are ready for review, could you take a look? Thanks!

File: chapter_computer-vision/image-augmentation.md [hunk 3]
```diff
@@ -431,6 +431,7 @@ def train_with_data_aug(train_augs, test_augs, net, lr=0.001):
     test_iter = load_cifar10(False, test_augs, batch_size)
     loss = nn.CrossEntropyLoss(reduction="none")
     trainer = torch.optim.Adam(net.parameters(), lr=lr)
+    net(next(iter(train_iter))[0]) # A dry run to initialize the model
```
Comment (astonzhang) [inline chapter_computer-vision/image-augmentation.md:434]: Add two space before an inline comment

File: chapter_computer-vision/kaggle-cifar10.md [hunk 4]
```diff
@@ -525,6 +525,7 @@ train(net, train_iter, valid_iter, num_epochs, lr, wd, devices, lr_period,
 #@tab pytorch
 devices, num_epochs, lr, wd = d2l.try_all_gpus(), 20, 2e-4, 5e-4
 lr_period, lr_decay, net = 4, 0.9, get_net()
+net(next(iter(train_iter))[0]) # A dry run to initialize the model
```
Comment (astonzhang) [inline chapter_computer-vision/kaggle-cifar10.md:528]: ditto

File: chapter_computer-vision/kaggle-cifar10.md [hunk 5]
```diff
@@ -554,6 +555,7 @@ df.to_csv('submission.csv', index=False)
 ```{.python .input}
 #@tab pytorch
 net, preds = get_net(), []
+net(next(iter(train_valid_iter))[0]) # A dry run to initialize the model
```
Comment (astonzhang) [inline chapter_computer-vision/kaggle-cifar10.md:558]: ditto

File: chapter_computer-vision/image-augmentation.md (hunk 3 above)
Comment (astonzhang) [reply chapter_computer-vision/image-augmentation.md:434]: In this case it's a bit verbose so let's remove the comment

**Review by astonzhang — APPROVED** [review]
Thanks! Please take a look at my minor comments. I'm merging it and revising it in https://github.com/d2l-ai/d2l-en/commit/b928a7f1218345f3988f4c86b696370f1eebf850.

**Pull request merged**